### PR TITLE
rls: minor config processing changes

### PR DESCRIPTION
**Pull request by easwars**
- enforce a max cache size of 5MB
- 64-bit align the cacheSizeBytes field in the lbConfig struct
- minor changes to comments and error messages

RELEASE NOTES: n/a

### CODE REVIEW

File: balancer/rls/internal/config.go [hunk 1]
```diff
@@ -109,37 +105,49 @@ type lbConfigJSON struct {
...
+//   - if `stale_age` is set, then `max_age` must also be set
+//   - ignore `valid_targets` field
+//   - `cache_size_bytes` field must have a value greater than 0, and if its
+//      value is greater than 5M, we cap it at 5M
+// - childPolicy:
+//   - must find a valid child policy with a valid config
+// - childPolicyConfigTargetFieldName:
 //   - must be set and non-empty
-func (*rlsBB) ParseConfig(c json.RawMessage) (serviceconfig.LoadBalancingConfig, error) {
+func parseConfig(c json.RawMessage) (serviceconfig.LoadBalancingConfig, error) {
```
Comment (dfawley) [inline balancer/rls/internal/config.go:133]: Why this change?  It seems to complicate this control flow unnecessarily by adding in an extra call into the stack.

File: balancer/rls/internal/builder.go [hunk 2]
```diff
@@ -43,3 +47,8 @@ func (*rlsBB) Build(cc balancer.ClientConn, opts balancer.BuildOptions) balancer
 	// TODO(easwars): Fix this once the LB policy implementation is pulled in.
 	return &rlsBalancer{}
 }
+
+func (*rlsBB) ParseConfig(c json.RawMessage) (serviceconfig.LoadBalancingConfig, error) {
```
Comment (dfawley) [inline balancer/rls/internal/builder.go:51]: Not a major deal but we tend to use non-pointer receivers for balancer builders (and register the struct instead of a pointer to a struct instance).  AIUI it's slightly more optimized that way due to memory allocations.

**Review by dfawley — APPROVED** [review]

File: balancer/rls/internal/builder.go (hunk 2 above)
Comment (easwars) [reply balancer/rls/internal/builder.go:51]: Done.

File: balancer/rls/internal/config.go (hunk 1 above)
Comment (easwars) [reply balancer/rls/internal/config.go:133]: The builder does not have any state that is used in this function, hence I thought there is no need for this function to have a receiver. Also, we do have usages like this currently in the codebase (although that is not a good enough reason).

Also, I find it easier to navigate through the code when a single type's methods are not spread across multiple files. Splitting the config parsing logic this way helps put all the config parsing logic in one file, and the builder implementation is just boiler plate.

Also, wouldn't the compiler be able to optimize this and inline the function?

Comment (dfawley) [reply balancer/rls/internal/config.go:133]: > The builder does not have any state that is used in this function, hence I thought there is no need for this function to have a receiver.

This is when you omit the variable for the receiver, though.

> Also, I find it easier to navigate through the code when a single type's methods are not spread across multiple files. Splitting the config parsing logic this way helps put all the config parsing logic in one file, and the builder implementation is just boiler plate.

Adding more code and more functions to make it easier to navigate?  I think naming the file `config.go` is probably good enough.  Or move the implementation of `ParseConfig` into `builder.go` as the other option?

> Also, wouldn't the compiler be able to optimize this and inline the function?

I'm not really concerned about performance.  We don't parse configs very often.  It just seems unnecessary to have this function here do ~nothing and call another function in the same package.  If you want to signal that the builder implements it, add a `var _ balancer.ConfigParser = rlsBB{}` line near the `rlsBB` struct definition?

Comment (easwars) [reply balancer/rls/internal/config.go]: Done.

**Review by dfawley — APPROVED** [review]
